File: frontend/lib/models/vehicle.dart
enum VehicleType {
  truck('truck'),
  trailer('trailer');

  const VehicleType(this.value);
  final String value;
}

class Vehicle {
  const Vehicle({
    required this.id,
    required this.type,
    required this.clientName,
    this.registrationNumber,
    this.chassisNumber,
    this.serialNumber,
    this.vin,
    this.clientContact,
    this.initialInspection,
  });

  final int id;
  final VehicleType type;
  final String clientName;
  final String? registrationNumber;
  final String? chassisNumber;
  final String? serialNumber;
  final String? vin;
  final String? clientContact;
  final String? initialInspection;

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as int,
      type: VehicleType.values.firstWhere(
        (type) => type.value == json['type'],
        orElse: () => VehicleType.truck,
      ),
      clientName: json['client_name'] as String,
      registrationNumber: json['registration_number'] as String?,
      chassisNumber: json['chassis_number'] as String?,
      serialNumber: json['serial_number'] as String?,
      vin: json['vin'] as String?,
      clientContact: json['client_contact'] as String?,
      initialInspection: json['initial_inspection'] as String?,
    );
  }
}

File: backend/models/inventory/vehicle.py
from sqlalchemy import Column, Integer, String, ForeignKey, Enum, DateTime, Text
from sqlalchemy.orm import relationship
from ..user import User
from ...database import Base
import enum
from datetime import datetime

class VehicleType(enum.Enum):
    truck = "truck"
    trailer = "trailer"

class Vehicle(Base):
    __tablename__ = "vehicles"

    id = Column(Integer, primary_key=True, index=True)
    type = Column(Enum(VehicleType), nullable=False)
    registration_number = Column(String, index=True, nullable=True)
    chassis_number = Column(String, index=True, nullable=True)
    serial_number = Column(String, index=True, nullable=True)
    vin = Column(String, index=True, nullable=True)
    client_name = Column(String, nullable=False)
    client_contact = Column(String, nullable=True)
    intake_time = Column(DateTime, default=datetime.utcnow)
    initial_inspection = Column(Text, nullable=True)
    admitted_by_id = Column(Integer, ForeignKey("users.id"))
    admitted_by = relationship("User")
